firedrake/functionspaceimpl.py

```diff
@@ -149,10 +149,17 @@ def topological(self, val):
         self.cargo.topological = val
 
     @utils.cached_property
-    def subfunctions(self):
+    def subspaces(self):
         r"""Split into a tuple of constituent spaces."""
         return tuple(type(self).create(subspace, self.mesh())
-                     for subspace in self.topological.subfunctions)
+                     for subspace in self.topological.subspaces)
+
+    @property
+    def subfunctions(self):
+        import warnings
+        warnings.warn("The 'subfunctions' property is deprecated, please use the "
```

Comment:
```suggestion
        warnings.warn("The 'subfunctions' property is deprecated for 'FunctionSpace', please use the "
```

Reply:
I tweaked the change a tiny bit, hope that's OK

Reply:
Sure, that's fine. It's just so that when people see it popping up they don't think they need to change all instances of `subfunctions`, only the ones for function spaces.